src/crimson/osd/osd.cc

```diff
@@ -364,13 +365,13 @@ void OSD::store_map_bl(ceph::os::Transaction& t,
                        epoch_t e, bufferlist&& bl)
 {
   meta_coll->store_map(t, e, bl);
-  map_bl_cache[e] = std::move(bl);
+  map_bl_cache.insert(e, std::move(bl));
 }
 
 seastar::future<bufferlist> OSD::load_map_bl(epoch_t e)
 {
-  if (auto found = map_bl_cache.find(e); found != map_bl_cache.end()) {
-    return seastar::make_ready_future<bufferlist>(found->second);
+  if (auto found = map_bl_cache.find(e); found) {
```

Comment:
The new `::find` returns smart pointers so this should be OK. Initially I've got a bit of confusion and had to go one commit back as the similarity to STL containers suggested an iterator to me. Maybe let's use `bool` instead of `auto`?

Reply:
updated and repushed. i was tempted to put
```c++
if (std::optional<auto> found = map_bl_cache.find(e); found) {
```
but probably let's leave it for future.